agent/engine/docker_task_engine.go

```diff
@@ -568,10 +581,16 @@ func (engine *DockerTaskEngine) serialPull(task *api.Task, container *api.Contai
 	defer seelog.Debugf("Released ImagePullDeleteLock after pulling image - %s. Task: %v", container.Image, task)
 	defer ImagePullDeleteLock.Unlock()
 
-	pullStart := time.Now()
+	// Recrod the task pull_started_at timestamp
+	pullStart := engine.time().Now()
 	defer func(startTime time.Time) {
 		seelog.Infof("Finished pulling container %v in %s. Task: %v", container.Image, time.Since(startTime).String(), task)
 	}(pullStart)
+
+	ok := task.SetPullStartedAt(pullStart)
```

Comment:
Should we set `PullStartedAt` time here or before the actual `client.PullImage` call in `pullAndUpdateContainerReference` method ?

Reply:
I don't think so, since there is a lock inside the `serialPull` and `concurrentPull`, it would be more accurate to log here, otherwise if the pull is waiting for this lock, the actual `PullStartedAt` would be incorrect.